Gordon360/Services/RecIMServices/LeagueService.cs

```diff
@@ -0,0 +1,119 @@
+﻿using Gordon360.Models.CCT;
+using Gordon360.Models.CCT.Context;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.Extensions.Configuration;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+
+namespace Gordon360.Services.RecIMServices
+{
+    public class LeagueService : ILeagueService
+    {
+        private readonly CCTContext _context;
+
+        public LeagueService(CCTContext context)
+        {
+            _context = context;
+        }
+
+
+        public IEnumerable<League> GetAllLeagues()
+        {
+            var leagues = _context.League.AsEnumerable();
+            return leagues;
+        }
+        public League GetLeagueByID(int leagueID)
+        {
+            var result = _context.League
+                            .Where(l => l.ID == leagueID)
+                            .FirstOrDefault();
```

Comment:
FYI, there's an overload for `FirstOrDefault` that accepts a lambda to specify a filter clause. That is, you condense these two functions into one:
```suggestion
                            .FirstOrDefault(l => l.ID == leagueID);
```

That will behave identically.

Reply:
I had no clue! I'll make that adjustment (I'm sure it occurs more than once)